### ISSUE TITLE
Benchmarking for set operations via Numpy

### ISSUE THREAD

**Issue by Demirrr** [dice-group/Ontolearn#14]
Let `A` and `B` be a DL concept. To construct a new concept that is their intersection, we perform `AND` set operation on `A.instances` and `B.instances`. Note that instances field corresponds to set of owlready2 instances objects. If we convert a python set of owlready2 instances into numpy arrays, then we could make use of many other libraries that can work on numpy arrays. For instance
- sklearn [metrics](https://scikit-learn.org/stable/modules/model_evaluation.html#classification-metrics) at calculation of accuracy, F-1 score.
- Rule learning algorithms provided in [Mlxtend](http://rasbt.github.io/mlxtend/user_guide/frequent_patterns/apriori/)

What we could do 

1. Represent instances as natural numbers. To this end, Let f be a mapping  such that f :  owlready2 instance -> N^+. Following store all integers as numpy array and use [this](https://numpy.org/doc/stable/reference/generated/numpy.intersect1d.html) for intersection.
